src/Registry.php

```diff
@@ -827,4 +840,86 @@ protected function toFlatten($separator = null, $data = null, array &$array = []
             $array[$key] = $v;
         }
     }
+
+    /**
+     * Magic method to access separator property.
+     *
+     * @param  string  $name  The name of the property.
+     *
+     * @return string|null A value if the property name is valid, null otherwise.
+     *
+     * @since       __DEPLOY_VERSION__
+     * @deprecated  4.0  This is a B/C proxy for deprecated read accesses
+     */
+    public function __get($name)
+    {
+        switch ($name) {
+            case 'separator':
+                \trigger_deprecation(
+                    'joomla/registry',
+                    '__DEPLOY_VERSION__',
+                    'The $separator parameter will be removed in version 4.',
+                    self::class,
+                    self::class
+                );
+
+                return $this->separator;
+
+            default:
+                $trace = \debug_backtrace();
+                \trigger_error(
+                    \sprintf(
+                        'Undefined property via __get(): %1$s in %2$s on line %3$s',
+                        $name,
+                        $trace[0]['file'],
+                        $trace[0]['line']
+                    ),
+                    E_USER_NOTICE
+                );
+
+                return null;
+        }
+    }
+
+    /**
+     * Magic method to access separator property.
+     *
+     * @param  string  $name   The name of the property.
+     * @param  mixed   $value  The value of the property.
+     *
+     * @return void
+     *
+     * @since       __DEPLOY_VERSION__
+     * @deprecated  4.0  This is a B/C proxy for deprecated read accesses
+     */
+    public function __set($name, $value)
+    {
+        switch ($name) {
+            case 'separator':
+                \trigger_deprecation(
+                    'joomla/registry',
+                    '__DEPLOY_VERSION__',
+                    'The $separator parameter will be removed in version 4.',
+                    self::class,
+                    self::class
+                );
+
+                $this->separator = $value;
+                break;
+
+            default:
+                if (property_exists($this, $name)) {
+                    throw new \RuntimeException(
+                        \sprintf(
+                            'Cannot access protected or private property %s::$%s',
+                            __CLASS__,
+                            $name
+                        )
+                    );
+                }
+
+                $this->$name = $value;
```

Comment:
Only the separator value is allowed, right? It seems like any value can be set with this method now. The registry is not receiving a magic setter method (actually), but we are using this path to deprecate public access to the `separator` value. We should not imply that we added the `__set` method to allow the setting of any other values in the registry class. Can you please update this to always give a `\RuntimeException(` if any other value is being set, than the `separator`.

Reply:
Should that be too bold for version 2, then use `\trigger_deprecation(` to inform users that this behavior will be removed in version 3. Currently, this behavior circumvents the `\trigger_error(` in the getter method, which gives the impression that it is a feature. However, we know that this is a side effect that will be removed once the setter is removed.

Reply:
I think PHP 8.2 should triger deprecation already for "dynamic properties".

Reply:
Makes sense. Added.